components/script/dom/domparser.rs

```diff
@@ -54,7 +55,7 @@ impl DOMParserMethods for DOMParser {
                        ty: DOMParserBinding::SupportedType)
                        -> Fallible<DomRoot<Document>> {
         let url = self.window.get_url();
-        let content_type = DOMString::from(ty.as_str());
+        let content_type = ty.as_str().parse().unwrap_or(Mime(TopLevel::Text, SubLevel::Html, vec![]));
```

Comment:
All of the supported types should be valid MIME types, so let's do this instead: `ty.as_str().parse().expect("Supported type is not a MIME type")`.